prime-angular-frontend/src/app/modules/site-registration/pages/care-setting-page/care-setting-page.component.ts

```diff
@@ -170,15 +173,41 @@ export class CareSettingPageComponent extends AbstractCommunitySiteRegistrationP
   }
 
   protected submissionRequest(): NoContent {
-    if (this.formState.json.careSettingCode !== CareSettingEnum.DEVICE_PROVIDER) {
-      this.siteFormStateService.deviceProviderFormState.clearIndividualDeviceProviders();
-    }
-    if (this.formState.json.careSettingCode === CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE) {
-      this.siteFormStateService.businessLicenceFormState.resetSiteId();
+
+    const sitePayload = this.siteFormStateService.json;
+    const request$ = this.siteResource.updateSite(sitePayload);
+
+
+    if (this.formState.json.careSettingCode === CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE
+      && this.vendorConfig.find(v => v.code === this.formState.vendorCode.value).name === 'CareConnect'
+    ) {
+      const data: DialogOptions = {
+        title: 'Save & Continue',
+        message: `I confirm I have contacted CareConnect.`,
+        actionText: 'Yes, I\'ve contacted CareConnect',
+        actionType: 'warn'
+      };
+
+      return this.dialog.open(ConfirmDialogComponent, { data })
+        .afterClosed()
+        .pipe(
+          exhaustMap((confirmation: boolean) => {
+            if (confirmation) {
+              return request$;
+            }
+            return EMPTY;
+          })
+        );
+    } else {
+      if (this.formState.json.careSettingCode === CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE) {
+        this.siteFormStateService.businessLicenceFormState.resetSiteId();
+      }
```

Comment:
Not sure if I understand the need for `resetSiteId` but looking at previous logic, shouldn't `resetSiteId` be executed even if user has selected CareConnect?

Reply:
Yes, it was missed. The code is updated. FYI, I believe the code to reset the site id is for case where the care setting changes from community pharmacy to PCHP.